Update Teams Shifts Connector Cmdlets to V2 APIs (#10741)
MicrosoftDocs/office-docs-powershell: teams/teams-ps/teams/Set-CsTeamsShiftsConnection.md

```

### SetViaIdentity
```
Set-CsTeamsShiftsConnection -InputObject <IConfigApiBasedCmdletsIdentity> -Body <IUpdateWfmConnectionRequest> [-Authorization <string>] [-IfMatch <string>] [-Break] [-HttpPipelineAppend <SendAsyncStep[]>] [-HttpPipelinePrepend <SendAsyncStep[]>] [-Proxy <uri>] [-ProxyCredential <pscredential>] [-ProxyUseDefaultCredentials] [-WhatIf] [-Confirm] [<CommonParameters>]
```

## DESCRIPTION
This cmdlet updates a Shifts WFM connection. It allows the admin to make changes to the settings such as the name and WFM URLs. Note that the update allows for, but does not require, the -ConnectorSpecificSettings.LoginPwd and ConnectorSpecificSettings.LoginUserName to be included.
This cmdlet can update every input field except -ConnectorId and -ConnectionId.

## EXAMPLES

> ### Example WFM 1

Reviewer: ```suggestion
### Example 1
```

ConnectorSpecificSettingAdminApiUrl      : https://www.contoso.com/retail/data/wfmadmin/api/v1-beta2
ConnectorSpecificSettingApiUrl           :
ConnectorSpecificSettingAppKey           :
ConnectorSpecificSettingClientId         :
ConnectorSpecificSettingCookieAuthUrl    : https://www.contoso.com/retail/data/login
ConnectorSpecificSettingEssApiUrl        : https://www.contoso.com/retail/data/wfmess/api/v1-beta2
ConnectorSpecificSettingFederatedAuthUrl : https://www.contoso.com/retail/data/login
ConnectorSpecificSettingRetailWebApiUrl  : https://www.contoso.com/retail/data/retailwebapi/api/v1
ConnectorSpecificSettingSiteManagerUrl   : https://www.contoso.com/retail/data/wfmsm/api/v1-beta2
ConnectorSpecificSettingSsoUrl           :
CreatedDateTime                          : 24/03/2023 04:58:23
Etag                                     : "5b00dd1b-0000-0400-0000-641d2df00000"
Id                                       : 4dae9db0-0841-412c-8d6b-f5684bfebdd7
LastModifiedDateTime                     : 24/03/2023 04:58:23
Name                                     : Cmdlet test connection - updated
TenantId                                 : 3FDCAAF2-863A-4520-97BA-DFA211595876

```

> Updates the instance with the specified -ConnectionId. Returns the object of updated connection.

Reviewer 2: ```suggestion
Updates the instance with the specified -ConnectionId. Returns the object of the updated connection.
```

> In case of error, we can capture the error response as following:

Reviewer 2: ```suggestion
In case of an error, you can capture the error response as follows:
```

* Hold the cmdlet output in a variable: `$result=<CMDLET>`

* To get the entire error message in Json: `$result.ToJsonString()`

* To get the error object and object details: `$result, $result.Detail`

> ### Example WFM 2

Reviewer: ```suggestion
### Example 2
```

            LoginPwd = "PlaceholderForPassword"
    })
PS C:\> $result | Format-List
```

```output

ConnectorId                              : 95BF2848-2DDA-4425-B0EE-D62AEED4C0A0
ConnectorSpecificSettingAdminApiUrl      :
ConnectorSpecificSettingApiUrl           : https://www.contoso.com/api
ConnectorSpecificSettingAppKey           :
ConnectorSpecificSettingClientId         : Es5Q2fB4PXweCyto6Ms6J2kRB7uwAc3W
ConnectorSpecificSettingCookieAuthUrl    :
ConnectorSpecificSettingEssApiUrl        :
ConnectorSpecificSettingFederatedAuthUrl :
ConnectorSpecificSettingRetailWebApiUrl  :
ConnectorSpecificSettingSiteManagerUrl   :
ConnectorSpecificSettingSsoUrl           : https://www.contoso.com/sso
CreatedDateTime                          : 06/04/2023 11:05:39
Etag                                     : "3100fd6e-0000-0400-0000-642ea7840000"
Id                                       : a2d1b091-5140-4dd2-987a-98a8b5338744
LastModifiedDateTime                     : 06/04/2023 11:05:39
Name                                     : Cmdlet test connection - updated
TenantId                                 : 3FDCAAF2-863A-4520-97BA-DFA211595876

```

> Updates the instance with the specified -ConnectionId. Returns the object of updated connection.

Reviewer 2: ```suggestion
Updates the instance with the specified -ConnectionId. Returns the object of the updated connection.
```

> In case of error, we can capture the error response as following:

Reviewer 2: ```suggestion
In case of an error, you can capture the error response as follows:
```

* Hold the cmdlet output in a variable: `$result=<CMDLET>`

* To get the entire error message in Json: `$result.ToJsonString()`

* To get the error object and object details: `$result, $result.Detail`

## PARAMETERS

### -Body

> The request body 

Reviewer 2: ```suggestion
The request body.
```

> The request body 

Reviewer 2: ```suggestion
The request body.
```

```yaml
Type: IUpdateWfmConnectionRequest
Parameter Sets: Set, SetViaIdentity
Aliases:

Required: True
Position: Named
Default value: None
Accept pipeline input: True (ByValue)
Accept wildcard characters: False
```

### -Break
Wait for .NET debugger to attach.

```yaml
Type: SwitchParameter
Parameter Sets: (All)
Aliases:

Required: False
Position: Named
Default value: None
Accept pipeline input: False
Accept wildcard characters: False
```

### -Confirm
Prompts you for confirmation before running the cmdlet.

```yaml
Type: SwitchParameter
Parameter Sets: (All)
Aliases: cf

Required: False
Position: Named
Default value: None
Accept pipeline input: False
Accept wildcard characters: False
```

### -ConnectionId

> The WFM connection id for the instance. This can be retrieved by running [Get-CsTeamsShiftsConnection](Get-CsTeamsShiftsConnection.md).

Reviewer 2: ```suggestion
The WFM connection ID for the instance. This can be retrieved by running [Get-CsTeamsShiftsConnection](Get-CsTeamsShiftsConnection.md).
```
